### DISCUSSION TITLE
Vector tiles in react-native-maps

### DISCUSSION THREAD

**Discussion by far1din** [react-native-maps/react-native-maps#4499] [Q&A]
I want to add a custom vector tile to the map, but can't seem to figure out how. Is there any way to add vector/raster tiles with this package? If not, is there a workaround?

**Comment by muzammilideofuzion** [comment]
You can simply add them using UrlTile/WmsTile.

**Comment by Haseeba393** [comment]
You can use UrlTile/WmsTile to show vector/raster tiles on map. You can also find example for both [https://github.com/react-native-maps/react-native-maps/tree/master/example/src/examples](url)

**Reply by far1din** [reply]
Thank you sir for the reply. I am very new to this and don't know much. 

Where can I find other tiles in the format suggested in the docs (http://c.tile.openstreetmap.org/{z}/{x}/{y}.png) and is it possible to customize and download?

There also seem to be a problem when using this tile layer as it disappears when zooming in too much. Do you know what the problem might be?

**Reply by Haseeba393** [reply]
Your welcome @far1din.
Brother in docs they are using one of the <a href="https://www.openstreetmap.org">Open Street Map's Layer</a>. You have to make an account for <a href="https://www.openstreetmap.org">Open Street</a> and there you will get multiple layers options, with own map generator and other formats that Open Street supports. You can also checkout following maps provider as well:

1. <a href="https://www.maptiler.com/">Map Tiler</a>
2. <a href="https://www.aerisweather.com/">Aeris Weather</a>

and the question you have mentioned about tiles disappearing on zoom, that is the native behaviour of the maps.

**Reply by far1din** [reply]
My friend, thank you so much for the answer. This helped me out a lot!! 💯

**Reply by Haseeba393** [reply]
Your most welcome
